### PR TITLE
Regression test look for extension crashes caused by entering more than 6 decimal places

### PR DESCRIPTION
**Pull request by timstackblock**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1243266896).

This is a regression test to look for web wallet crashes that are caused by entering more than 7 decimal places. The test attempts to reproduce the P1 bug we found in issue #1698 as well as additional test that test edge cases related to that functionality.

@UXTEAM
cc/ @aulneau @kyranjamie @fbwoolf @beguene @markmhx @Eshwari007 @andresgalante

### CODE REVIEW

**Comment by timstackblock** [comment]
I will fix these linting issues reformatted but prettier did not take.

File: README.md [hunk 1]
```diff
@@ -163,3 +163,4 @@ Please note this email is strictly for reporting security vulnerabilities. For s
 In Q1 2021, Hiro partnered with [Least Authority](https://leastauthority.com/), a leading security consultancy with experience in the crypto space, to audit Hiro Wallet for Web. On April 29th 2021, after addressing the major concerns described in the initial findings, as well as a concluding sign off from the Least Authority team, a final report was delivered.
 
 [Download and read the full report here](https://github.com/blockstack/stacks-wallet-web/blob/main/public/docs/least-authority-security-audit-report.pdf)
+timtest123456
```
Comment (kyranjamie) [inline README.md:166]:
```suggestion
```

File: src/pages/transaction-signing/components/event-card.tsx [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
+                                                      title,
+                                                      left,
+                                                      right,
+                                                      amount,
+                                                      ticker,
+                                                      icon,
+                                                      message,
+                                                      actions,
+                                                      isLast,
+                                                    }) => {
```
Comment (kyranjamie) [inline src/pages/transaction-signing/components/event-card.tsx:19]: Make sure you're using the prettier plugin for your IDE. This is illegal formatting. I have it set up so that when I save,  it formats automatically.

File: src/pages/transaction-signing/components/event-card.tsx [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
   return (
     <>
       <Stack spacing="base-loose" p="base-loose">
         <SpaceBetween position="relative">
           <Text fontWeight={500} fontSize={2}>
-            {title}
+            <span data-testid={SendFormSelectors.TransferMessage}>{title}</span>
```
Comment (kyranjamie) [inline src/pages/transaction-signing/components/event-card.tsx:25]: It isn't necessary to create a new `span` here. The `data-testid` selector can go on `Text`

File: src/pages/transaction-signing/components/event-card.tsx [hunk 4]
```diff
@@ -1,34 +1,35 @@
...
           </Text>
-          {actions && <IconButton size="24px" icon={IconDots} position="absolute" right={0} />}
+          {actions && <IconButton size="24px" icon={IconDots} position="absolute" right={0}/>}
```
Comment (kyranjamie) [inline src/pages/transaction-signing/components/event-card.tsx:27]: More formatter issues. Try and configure your IDE properly, lmk if you need help.

**Review by kyranjamie — COMMENTED** [review]
Good work @timstackblock—

There's some tidying up to do re line edits. Many lines changed that are purely formatting.

I wonder if there's a way to test this functionality, but without checking the exact error string. If we change the wording it'll fail, which isn't super helpful. Do you have any ideas how we might do this?

I'd note that that error with the app crashing wasn't directly related to the decimals. You could trigger any error and it'd crash on 2nd validation attempt.

Nonetheless, these are helpful tests 👍🏼   When you're done making changes, happy to run you through how to rebase your commits in to a single one so that `timtest1` doesn't make it into the history lol.

File: tests/integration/send-tokens/send-tokens.spec.ts [hunk 5]
```diff
@@ -54,12 +56,87 @@ describe(`Send tokens flow`, () => {
...
+  }, 30_000);
+
+  afterEach(async () => {
+    try {
+      await browser.context.close();
+    } catch (error) {
+    }
+  });
+
+  it('expect that the preview is shown', async () => {
```
Comment (fbwoolf) [inline tests/integration/send-tokens/send-tokens.spec.ts:118]: I think these statements might read better like, `it('should show the preview',...`

Same for one after, `it('should show the preview when there...`

**Comment by fbwoolf** [comment]
Looks great @timstackblock except for the formatting issues Kyran pointed out.

File: src/pages/transaction-signing/components/event-card.tsx (hunk 2 above)
Comment (timstackblock) [reply src/pages/transaction-signing/components/event-card.tsx:19]: Yeah I thought I had Webstorm IDE setup to run prettier with a keyboard short cut. It reformatted but didn't include prettier I can fix this.

File: README.md (hunk 1 above)
Comment (timstackblock) [reply README.md:166]: This wasn't suppose to be included in the the PR I was testing my github ssh connection in a prior branch. For some reason some commits come up as TW without my avatar, I fixed the issue but the test commit got pulled into this PR. I can fix this as well.

Comment (timstackblock) [reply README.md:166]: > I wonder if there's a way to test this functionality, but without checking the exact error string. If we change the wording it'll fail, which isn't super helpful. Do you have any ideas how we might do this?
> 

The quickest way to check for the assertion is with the string, it gets the job done and gives us some protection from this P1 error coming back in a regression which was my first concern but we can definitely improve these test in iterations  with multiple PRs as we add more test.
